.circleci/config.yml

```diff
@@ -1,15 +1,38 @@
 version: 2
 jobs:
+  generate_cache_for_system_test:
+    machine: true
+    steps:
+      - checkout
+      - restore_cache:
+          key: docker-{{ checksum ".circleci/config.yml" }}-{{ checksum "docker-compose.system-test.yml" }}-{{ checksum "Dockerfile.system-test" }}
+          paths: ~/caches/images.tar
+      - run:
+          command: |
+            if [ ! -f ~/caches/images.tar ]; then
+              docker-compose -f docker-compose.system-test.yml build && \
+              mkdir -p ~/caches && \
+              docker save $(docker images | awk 'NR>=2 && ! /^<none>/{print $1}') -o ~/caches/images.tar
+            fi
+      - save_cache:
+          key: docker-{{ checksum ".circleci/config.yml" }}-{{ checksum "docker-compose.system-test.yml" }}-{{ checksum "Dockerfile.system-test" }}
+          paths: ~/caches/images.tar
   system_test:
     machine: true
+    parallelism: 2
     steps:
       - checkout
+      - restore_cache:
+          key: docker-{{ checksum ".circleci/config.yml" }}-{{ checksum "docker-compose.system-test.yml" }}-{{ checksum "Dockerfile.system-test" }}
+          paths: ~/caches/images.tar
```

Comment:
I think you don't need to use paths in here. 
https://circleci.com/docs/2.0/configuration-reference/#restore_cache

Reply:
Fixed.